src/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.Unix.cs

```diff
@@ -9,42 +14,195 @@ private FileVersionInfo(string fileName)
+                                if (_productVersion == "0.0.0.0")
+                                {
+                                    _productVersion = _fileVersion;
+                                    _productMajor = _fileMajor;
+                                    _productMinor = _fileMinor;
+                                    _productBuild = _fileBuild;
+                                    _productPrivate = _filePrivate;
+                                }
+                            }
+                            break;
+                        case "System.Reflection.AssemblyProductAttribute":
+                            _productName = attrNameAndValue.Value;
+                            break;
+                        case "System.Reflection.AssemblyTrademarkAttribute":
+                            _legalTrademarks = attrNameAndValue.Value;
+                            break;
+                        case "System.Reflection.AssemblyTitleAttribute":
+                            _fileDescription = attrNameAndValue.Value;
+                            break;
+                    }
+                }
+            }
         }
 
-        // -----------------------------
-        // ---- PAL layer ends here ----
-        // -----------------------------
+        /// <summary>
+        /// Attempt to parse a CustomAttribute into its full name and the value of its single string parameter.
+        /// </summary>
+        /// <param name="reader">The metadata reader.</param>
+        /// <param name="attr">The attribute to parse.</param>
+        /// <param name="result">The name of the attribute and its parameter's value, if it could be parsed.</param>
+        /// <returns>true if the attribute could be parsed; otherwise, false.</returns>
+        private static bool TryGetAttributeNameAndStringArgument(
+            MetadataReader reader, CustomAttribute attr, out KeyValuePair<string, string> result)
+        {
+            var ctorHandle = attr.Constructor;
+            if (ctorHandle.Kind == HandleKind.MemberReference)
+            {
+                EntityHandle container = reader.GetMemberReference((MemberReferenceHandle)ctorHandle).Parent;
+                if (container.Kind == HandleKind.TypeReference)
+                {
+                    TypeReference attrType = reader.GetTypeReference((TypeReferenceHandle)container);
+                    string attrName = reader.GetString(attrType.Namespace) + "." + reader.GetString(attrType.Name);
+
+                    if (attr.Constructor.Kind == HandleKind.MemberReference)
+                    {
+                        var signature = reader.GetMemberReference((MemberReferenceHandle)attr.Constructor).Signature;
+                        var signatureReader = reader.GetBlobReader(signature);
+                        var valueReader = reader.GetBlobReader(attr.Value);
+
+                        if (valueReader.ReadUInt16() == 1)
```

Comment:
Consider using a const and/or comment to indicate that this magic two-byte 1 is what Ecma 335 (see II.23.3) calls the 'prolog' and it must be at the front of every valid attribute value blob.